### ISSUE TITLE
GG Max Bug Report - Standalone issue and bugs list

### ISSUE THREAD

**Issue by UltraVox001** [TheGameCreators/GameGuruRepo#1757]
Hi,

Some build tests done tonight to test the .exe quality, and then produce a full bug report, based on multiple tests. Here is the list of problems observed. Note that these problems are systematic. I recompiled the project 5 times. All the problems noted are on the same 3D objects. There is no such thing as "chance" in placement bugs. It was only certain objects, but never the others.

1/ Sound problems.
- The sound of the atmosphere (plains.wav) does not work.
- The sound produced by the player's footsteps does not work. No soil texture produces the sound of footsteps.
- The house does not produce the sound of footsteps on the wood.

Last week the ground textures didn't make any sound, but the wooden house made a sound of footsteps on wood.

- The sound produced by the opening and closing of the doors, works.
- The sound produced by all the switches, works.
- The sound produced by firing and reloading a weapon, works.

2/ Problems of placement with 3D objects.
Some 3D objects are moved several meters in the map.
![od1](https://user-images.githubusercontent.com/96718175/149431293-8e11f27b-7919-4a4f-8c12-0288b9dbea7e.jpg)
![od1-editor](https://user-images.githubusercontent.com/96718175/149431295-4f58214c-fbd4-4442-8d4c-34fd223f2a24.jpg)
They are all in Static mode, with Box collisions, or without collisions.
This chimney flue is moved to the next room. On the lower floor, the fireplace no longer has a flue.
![od2-editor](https://user-images.githubusercontent.com/96718175/149431356-c4861554-ab6b-4265-ac32-bb00d162de32.jpg)
![od2-2](https://user-images.githubusercontent.com/96718175/149431368-1f1ce1b2-716b-4844-806a-90eec54dac33.jpg)
![od2](https://user-images.githubusercontent.com/96718175/149431324-ef171dd1-b77d-4b06-80db-cb5173aedc69.jpg)

Out of 614 3D objects, only one has completely disappeared (not found) from the map in the compiled version. He too is Static with Box collision.
![disparu](https://user-images.githubusercontent.com/96718175/149431391-4f597c97-2ea6-4813-96f8-d6e967af07e1.jpg)
![disparu-editor](https://user-images.githubusercontent.com/96718175/149431398-de3d1bde-cdf1-43bd-acdf-df1c4b82ea4a.jpg)

Hoping it can help.

**Comment by rvanner** [comment]
@UltraVox001 Thanks for the extensive tests and report. Would you be able to send me your test map so we can use it to find the issues you have reported here?

**Comment by UltraVox001** [comment]
Yes, naturally.
I sent you a download link on the forum, in a private message.

Thank you and good day.

**Comment by UltraVox001** [comment]
Hi,

I found the reason why the 3D object "CoffeeTable_05a" is systematically missing from the executable version.

See this screenshot :
![CoffeeTable](https://user-images.githubusercontent.com/96718175/149620099-edbdc565-1ad7-4e0e-ba1b-d2242bbb03d4.jpg)
For some reason, this object does not appear (it has disappeared) in the list of 3D objects, located on the left of Max. I noticed this while trying to solve the problem found in the executable. I deleted this table in the editor, then looked for it in the list, but it's not there. So Max couldn't integrate it into the compilation.

It remains to be seen why an object added in 3D space would not be memorized in this list, since it is present in the level.

**Comment by UltraVox001** [comment]
I did a new test in an empty map.

I put the table on the floor.
It appears in the list of objects.
![editor](https://user-images.githubusercontent.com/96718175/149620627-87507f8d-d673-43d6-b143-a1f43a40bf7c.jpg)
I'm doing a testGame.
The table appears in the TestGame.
I save, and I compile.
The table does not appear in the compiled version.
![exe](https://user-images.githubusercontent.com/96718175/149620637-0cd943b1-5ffd-4c5d-ada9-bce62447336a.jpg)
However, it appears in the list of objects when I reload the map.

Strange.

**Comment by UltraVox001** [comment]
This problem seems to come from this transparent table.
When I place another one, from the same 3D pack, it works perfectly in the executable.
![exe](https://user-images.githubusercontent.com/96718175/149620921-42f4952b-7ea2-4e0d-bb16-bd8fd4a46b80.jpg)

**Comment by UltraVox001** [comment]
I also did a Standalone test to understand why the objects I use to simulate a chimney flue are never placed correctly in the executable version.

It turns out that this problem is related to the 3D objects themselves.
I use a table as a 3D reference. The other two objects must be located on each side of the table.
In an empty map, everything is fine in the editor :
![editor](https://user-images.githubusercontent.com/96718175/149622620-271cf063-1cec-447f-9f01-9adbdca83be7.jpg)
All is well in TestLevel :
![TestLevel](https://user-images.githubusercontent.com/96718175/149622654-41134ead-edf3-490f-be38-66b6f89313fa.jpg)
But in the executable, it doesn't work anymore :
![exe](https://user-images.githubusercontent.com/96718175/149622658-e62f39a4-a1e5-43e1-a6c2-d70f841a41e9.jpg)
I hope it can help.

**Comment by rvanner** [comment]
@UltraVox001 Thanks for the detailed reporting, it will really help us get things like this fixed. We have 2 more weeks of final changes to the software and then we'll be focusing on the bugs.

**Comment by UltraVox001** [comment]
Ok. I suggest you avoid working on the map I sent you. For some reason Max has duplicated all of the 3D objects under the map a few hundred yards from the house. As a result, all 3D objects are duplicated. The FPS loss was due to this. I cleaned everything. I also removed the few 3D objects that were causing issues with lights and shadows. These are essentially objects from very old 3D packs (FPSC, Reloaded...). I'm not sure it's relevant that you waste your time working on this. :)
